pkg/asset/machines/worker.go

```diff
@@ -162,3 +161,16 @@ func applyTemplateData(template *template.Template, templateData interface{}) []
 	}
 	return buf.Bytes()
 }
+
+func listFromMachineSets(objs []clusterapi.MachineSet) *metav1.List {
```

Comment:
I feel like there should be a more expressive way to do this other than having per-type `listFrom*` converters.  But I haven't been able to find an incantation that works ;).

Reply:
all the types impl `metav1.Object`, but https://golang.org/doc/faq#convert_slice_of_interface